tests/test_annotation_stores.py

```diff
@@ -0,0 +1,980 @@
+    x, y = xy
+    alpha = np.linspace(0, 2 * np.pi - (2 * np.pi / n_points), n_points)
+    rx = radius * (np.random.rand() + 0.5)
+    ry = np.random.uniform(*eccentricity) * radius - rx
+    x = rx * np.cos(alpha) + x + (np.random.rand(n_points) - 0.5) * noise
+    y = ry * np.sin(alpha) + y + (np.random.rand(n_points) - 0.5) * noise
+    boundary_coords = np.stack([x, y], axis=1).astype(int).tolist()
+
+    # Copy first coordinate to the end if required
+    if repeat_first:
+        boundary_coords = boundary_coords + [boundary_coords[0]]
+
+    # Swap direction
+    if direction.strip().lower() == "cw":
+        boundary_coords = boundary_coords[::-1]
+
+    polygon = Polygon(boundary_coords)
+
+    # Add random rotation
+    angle = np.random.rand() * 360
+    return affinity.rotate(polygon, angle, origin="centroid")
+
+
+def sample_predicate(props: Dict[str, Any]) -> bool:
+    """Simple example predicate function for tests."""
+    return props.get("class") == 123
+
+
+# Fixtures
+
+
+@pytest.fixture(scope="session")
+def cell_grid() -> List[Polygon]:
+    """Generate a grid of fake cell boundary polygon annotations."""
+    np.random.seed(0)
+    return [cell_polygon((i * 25, j * 25)) for i, j in np.ndindex(*GRID_SIZE)]
+
+
+@pytest.fixture(scope="session")
+def points_grid() -> List[Polygon]:
+    """Generate a grid of fake point annotations."""
+    np.random.seed(0)
+    return [Point((i * 25, j * 25)) for i, j in np.ndindex(*GRID_SIZE)]
+
+
+@pytest.fixture(scope="session")
+def sample_triangle() -> Polygon:
+    """Simple traingle polygon used for testing."""
+    return Polygon([(0, 0), (1, 1), (2, 0)])
+
+
+@pytest.fixture()
+def fill_store(cell_grid, points_grid):
+    """Factory fixture to fill stores with test data."""
+
+    def _fill_store(
+        store_class: AnnotationStore,
+        path: Union[str, Path],
+    ):
+        store = store_class(path)
+        annotations = [Annotation(cell) for cell in cell_grid] + [
+            Annotation(point) for point in points_grid
+        ]
+        keys = store.append_many(annotations)
+        return keys, store
+
+    return _fill_store
+
+
+# Generate Parameterized Tests
+
+
+def pytest_generate_tests(metafunc):
+    """Generate (parameterize) test scenarios.
+
+    Adapted from pytest documentation. For more information on
+    parameterized tests see:
+    https://docs.pytest.org/en/6.2.x/example/parametrize.html#a-quick-port-of-testscenarios
+    """
+    # Return if the test is not part of a class
+    if metafunc.cls is None:
+        return
+    idlist = []
+    argvalues = []
+    for scenario in metafunc.cls.scenarios:
+        idlist.append(scenario[0])
+        items = scenario[1].items()
+        argnames = [x[0] for x in items]
+        argvalues.append([x[1] for x in items])
+    metafunc.parametrize(argnames, argvalues, ids=idlist, scope="class")
+
+
+# Class Specific Tests
+
+
+def test_sqlitestore_compile_options():
```

Comment:
can we say sqlite_score?

Reply:
Why sqlite_score, the name of the class is SQLiteStore and PyTest convention is to use the lowercase in the test name.